apps/content/core/src/main/java/com/adobe/dx/content/marketo/models/MarketoCompFooter.java

```diff
@@ -0,0 +1,151 @@
+
+import static com.adobe.cq.xf.ExperienceFragmentsConstants.PN_FRAGMENT_PATH;
+import static com.day.cq.commons.jcr.JcrConstants.JCR_CONTENT;
+import static org.apache.commons.lang.StringUtils.EMPTY;
+
+import com.adobe.dx.content.marketo.models.internal.MarketoConfBasicInfo;
+import com.adobe.dx.utils.service.CloudConfigReader;
+import com.day.cq.wcm.api.Page;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.annotation.PostConstruct;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.AbstractResourceVisitor;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.OSGiService;
+import org.apache.sling.models.annotations.injectorspecific.ScriptVariable;
+import org.jetbrains.annotations.NotNull;
+
+@Model(adaptables = SlingHttpServletRequest.class)
+public class MarketoCompFooter {
+
+    @OSGiService
+    private CloudConfigReader cloudConfigReader = null;
+
+    @ScriptVariable
+    private Page currentPage = null;
+
+    private static final String CONFIG_NAME = "marketo-config";
+
+    private static final String MARKETO_FORM_ID = "marketoFormId";
+
+    private MarketoConfBasicInfo marketoConfBasicInfo;
+
+    private Set<String> marketoFormIds =  new HashSet<>();
+
+    @PostConstruct
+    private void init() {
+        marketoConfBasicInfo = cloudConfigReader.getContextAwareCloudConfigRes(
+            currentPage.getPath(), CONFIG_NAME, MarketoConfBasicInfo.class);
+        Set<String> marketoComponentTypes = new HashSet<>(Arrays
+            .asList(marketoConfBasicInfo.getMarketoComponentTypes()));
+        marketoFormIds = getMarketoFormIds(marketoComponentTypes, currentPage);
+    }
+
+    public Set<String> getMarketoFormIds() {
+        return marketoFormIds;
+    }
+
+    public MarketoConfBasicInfo getMarketoConfBasicInfo() {
+        return marketoConfBasicInfo;
+    }
+
+    // TODO All the code below this to be replaced with the generic service.
+    private Set<String> getMarketoFormIds(Set<String> marketoComponentTypes, Page currentPage) {
+        ComponentReferenceFinder visitor = new ComponentReferenceFinder(marketoComponentTypes);
+        visitor.accept(currentPage.getContentResource());
+        return visitor.getMatchingResources().stream()
+            .map(this::getMarketoFormId)
+            .filter(StringUtils::isNotEmpty).collect(Collectors.toSet());
+    }
+
+    private String getMarketoFormId(Resource resource) {
+        return resource.getValueMap().get(MARKETO_FORM_ID, EMPTY);
+    }
+
+    private static class ComponentReferenceFinder extends AbstractResourceVisitor {
```

Comment:
Please check with Sabya that wanted to do something about that (discussed on internal corp list :-)), to rediscuss what i said there:

1. we should propose a contribution of TypeIterator to dev@sling, where specific case of XF references are taken as a generic "proxy" references
2. waiting for the above have that code in a separate class, as we'll need it in other places

Reply:
There would be a different generic service that Jens will add. So have coupled this one inside the class.

Reply:
ok :-/